docs/basics/parsetrees.md

```diff
@@ -32,34 +38,283 @@ q)eval ((/;+);(til;(+;2;2)))
+```txt
+(function; arg 1; …; arg n)
+```
+
+where every item could itself be a parse tree. In this way we see that every action in q is essentially a function evaluation.
+
+
+## `eval` and `value`
+
+[`eval`](../ref/eval.md) can be thought of as the dual to `parse`. The following holds for all valid q statements (without side effects) put into a string. (Recall that `value` executes the command inside a string.)
+
+```q
+//a tautology (for all valid q expressions str)
+q)value[str]~eval parse str
+1b
+q)value["2+4*7"]~eval parse"2+4*7" //simple example
+1b
+```
+
+When passed a list, `value` applies the first item (which contains a function) to the rest of the list (the arguments).
+
+```q
+q)function[arg 1;..;arg n] ~ value(function;arg 1;..;arg n)
+1b
+```
+
+When `eval` and `value` operate on a parse tree with no nested parse trees they return the same result. However it is not true that `eval` and `value` are equivalent in general. `eval` operates on parse trees, evaluating any nested parse trees, whereas `value` operates on the literals.
+
+```q
+q)value(+;7;3)                  //parse tree, with no nested trees
+10
+q)eval(+;7;3)
+10
+q)eval(+;7;(+;2;1))             //parse tree with nested trees
+10
+q)value(+;7;(+;2;1))
+'type
+```
+```q
+q)value(,;`a;`b)
+`a`b
+q)eval(,;`a;`b)                 //no variable b defined
+'b
+q)eval(,;enlist `a;enlist `b)
+`a`b
+```
+
+
+## Variadic operators
+
+Many operators and some keywords in k and q are [variadic](glossary.md#variadic): they are overloaded so that the behavior of the operator changes depending on the number and type of arguments. In q (not k) the unary form of operators such as (`+`, `$`, `.`, `&` etc.) is disabled: keywords are provided instead.
 
-!!! tip "Use `parse` to reveal the functional form of a qSQL query"
+For example, in k the unary form of the `$` operator equates to the `string`
+keyword in q.
 
-> The result will often include [k code](exposed-infrastructure.md) but it is usually recognizable and you can use it in functional form. — _Q for Mortals_ §A.67
+```q
+q)k)$42
+"42"
+q)$42                  //$ unary form disabled in q
+'$
+q)string 42
+"42"
+```
+
+!!! info "A parenthesized variadic function applied prefix is parsed as its unary form."
+
+```q
+q)($)42
+"42"
+```
+
+A familiar example of a variadic function is the Add Over function `+/` derived by applying the Over iterator to the Add operator.
+
+```q
+q)+/[1000;2 3 4]    // +/ applied binary
+1009
+q)+/[2 3 4]         // +/ applied unary
+9
+q)(+/)2 3 4         // +/ applied unary
+9
+```
+
+In k, the unary form of an operator can also be specified explicitly by suffixing it with a colon.
+
+```q
+q)k)$:42
+"42"
+```
+
+`+:` is a unary operator; the unary form of `+`. We can see this in the parse tree:
+
+```q
+q)parse"6(+)4"
+6
+(+:;4)
+```
+
+The items of a `parse` result use k syntax. Since (most of) the q keywords are defined in the `.q` namespace, we can use dictionary reverse lookup to find the meaning.
```

Comment:
```suggestion
The items of a `parse` result use k syntax. Since (most of) the q keywords are defined in the `.q` namespace, you can use dictionary reverse lookup to find the meaning.
```